contracts/protocol/integration/wrap-v2/ERC4626WrapV2Adapter.sol

```diff
@@ -0,0 +1,111 @@
+/*
+    Copyright 2023 Index Cooperative
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+    SPDX-License-Identifier: Apache License, Version 2.0
+*/
+
+pragma solidity 0.6.10;
+pragma experimental "ABIEncoderV2";
+
+import { IERC4626 } from "../../../interfaces/external/IERC4626.sol";
+
+/**
+ * @title ERC4626WrapV2Adapter
+ * @author pblivin0x
+ *
+ * Wrap adapter for ERC-4626 Vaults that returns data for wraps/unwraps of tokens
+ */
+contract ERC4626WrapV2Adapter {
+
+    /* ============ External Getter Functions ============ */
+
+    /**
+     * Generates the calldata to wrap an underlying asset into a wrappedToken.
+     *
+     * @param _underlyingToken      Address of the component to be wrapped
+     * @param _wrappedToken         Address of the desired wrapped token
+     * @param _underlyingUnits      Total quantity of underlying units to wrap
+     * @param _to                   Address to send the wrapped tokens to
+     *
+     * @return address              Target contract address
+     * @return uint256              Total quantity of underlying units (if underlying is ETH)
+     * @return bytes                Wrap calldata
+     */
+    function getWrapCallData(
+        address _underlyingToken,
+        address _wrappedToken,
+        uint256 _underlyingUnits,
+        address _to,
+        bytes memory /* _wrapData */
+    )
+        external
+        view
+        returns (address, uint256, bytes memory)
+    {
+        IERC4626 vault = IERC4626(_wrappedToken);
+        require(vault.asset() == _underlyingToken, "Must be a valid token pair");
+
+        bytes memory callData = abi.encodeWithSignature(
+            "deposit(uint256,address)",
```

Comment:
If you choose to switch to 0.8 you could use  'encodeWithSelector'. Not sure, but this might save gas by precomputing the selector hash.

Reply:
updated to use `encodeWithSelector` here [aeda1df](https://github.com/IndexCoop/index-protocol/pull/11/commits/aeda1df34e3c7cc5c4a3983c6c3531b5723034b3)